src/ripple/basics/BasicConfig.h

```diff
@@ -40,15 +39,18 @@ using IniFileSections = std::map<std::string, std::vector<std::string>>;
 /** Holds a collection of configuration values.
     A configuration file contains zero or more sections.
 */
-class Section : public beast::unit_test::detail::const_container<
-                    std::map<std::string, std::string, boost::beast::iless>>
+class Section
 {
 private:
     std::string name_;
+    std::map<std::string, std::string> lookup_;
```

Comment:
Why not `std::unordered_map` instead? We'd get better performance on inserts and lookups. The drawback is that we could get inconsistent results when using the stream operator. If that's a deal breaker then we could add a sorting feature to the stream operator, which won't be terrible since performance isn't critical for that operation.

Reply:
I used `std::map` because we were previously inheriting from `beast::unit_test::detail::const_container<std::map...>`.  Leaving it as `std::map` was the smallest change that accomplished the goal of removing this dependency on `beast::unit_test::detail`.

That said, I think keeping `lookup_` as a `std::map` is a good conservative choice.  The `Section`'s container type is exposed to users through the `begin()` and `end()` accessors.  If we change `lookup_` to a `std::unordered_map`, then we'll need to audit all users of those iterators to make sure they are not depending on `lookup_` being pre-sorted.

That's a long way of saying that leaving it a `std::map` is the conservative change.  I'm not convinced the speed improvement from changing to a `std::unordered_map` justifies the additional work and risk.